CoffeeMachineSimulator/CoffeeMachineSimulator.Services/Services/CoffeeService.cs

```diff
@@ -48,5 +58,48 @@ private bool IsCoffeeValid(CoffeeModel model)
         {
             return model.Id != Guid.Empty && !string.IsNullOrEmpty(model.Name) && model.Price != 0.0f;
         }
+
+        public EspressoMachineModel GenerateNewEspressoMachine(string name, Guid espressoMachineId)
+        {
+            var generatedEspressoMachine = Builder<EspressoMachineModel>.CreateNew()
+                    .With(x=>x.Name=name)
+                    .With(x=>x.Id=espressoMachineId)
+                .Build();
+            return generatedEspressoMachine;
+        }
+
+        public void DrinkCoffee(Guid coffeeId)
+        {
+            if (coffeeId == Guid.Empty || coffeeId == null) throw new Exception("Please provide an ID!");
+            if (coffeeModels.FirstOrDefault(x => x.Id== coffeeId).Filled == false) throw new Exception("The coffee you are trying to drink has already been drank!");
+            coffeeModels.FirstOrDefault(x => x.Id == coffeeId).Filled = false;
+        }
+
+        public void UpdateCoffee(CoffeeModel coffeeToUpdate)
+        {
+            if (coffeeToUpdate == null) throw new Exception("The coffee you provided is null");
+            if (!IsCoffeeValid(coffeeToUpdate)) throw new Exception("The coffee you provided is not valid");
+            if (coffeeModels.All(x => x.Id != coffeeToUpdate.Id)) throw new Exception("There is no coffee with the ID you provided!");
+            coffeeModels.FirstOrDefault(x => x.Id == coffeeToUpdate.Id).Name = coffeeToUpdate.Name;
+            coffeeModels.FirstOrDefault(x => x.Id == coffeeToUpdate.Id).Price = coffeeToUpdate.Price;
+            coffeeModels.FirstOrDefault(x => x.Id == coffeeToUpdate.Id).Filled = coffeeToUpdate.Filled;
+            coffeeModels.FirstOrDefault(x => x.Id == coffeeToUpdate.Id).EspressoMachine = coffeeToUpdate.EspressoMachine;
+            coffeeModels.FirstOrDefault(x => x.Id == coffeeToUpdate.Id).Sweetness = coffeeToUpdate.Sweetness;
```

Comment:
please add "coffeeModels.FirstOrDefault(x => x.Id == coffeeToUpdate.Id)" in a variable and set that variable properties